features/0037-present-proof/README.md

```diff
@@ -35,6 +35,32 @@ Diagrams in this protocol were made in draw.io. To make changes:
 - export the picture and HTML to your local copy of this repo, and
 - submit a pull request.
 
+### States
+
+#### states for Verifier
+
+* request-sent
+* proposal-received
+* request-received
+* presentation-received
+* done
+
+#### states for Prover
+
+* request-received
+* proposal-sent
+* presentation-sent
+* reject-sent
+* done
+
+For the most part, these states map onto the transitions shown in the choreography diagram in obvious ways. However, a few subtleties are worth highlighting:
+
+* The final transitions for the Verifier (Send Presentation Reject, Send Presentation Ack, and Receive Presentation Reject) all result in a final `done` state, but this `done` may or may not be the Verifier's intended outcome. We may want to tweak this in a future version of the protocol.
+* A similar situation exists with the final transitions for the Prover and its final transitions (Send Presentation Reject, Receive Presentation Ack, and Receive Presentation Reject).
+* When a Prover makes a (counter-)proposal, it transitions to the `proposal-sent` state. This state is only present by implication in the choreography diagram; it essentially equates to the null or begin state in that the Prover does nothing until a presentation request arrives, triggering the leftmost transition for the Prover.
```

Comment:
I believe an intended feature of the "proposal" message is that the prover can initiate the protocol using that message. In that case, the last bullet point would not apply, and there would be both "proposal-sent" and "proposal-received" states.